### PR TITLE
Feature/Update-logic-for-checking-if-NCDs-are-controlled-or-not-based-on-the-new-data-points-that-were-added-from-DWAPI

### PR DESCRIPTION
**Pull request by nobert-mumo**
Jira ticket:  

Mostly changed the FactNCD to adapt it to use the new column from CT_ChronicIllness

### CODE REVIEW

File: Scripts/NDWH/C&T FACT TABLES/load_FactNCDs.sql [hunk 1]
```diff
@@ -54,7 +54,7 @@ with ncd_source_data as (
                     "Thyroid disease"
                 )
             ) as pivot_table
-    where voided = 0
+    where voided = 0 and PatientPKHash is not null
```
Comment (nobert-mumo) [inline Scripts/NDWH/C&T FACT TABLES/load_FactNCDs.sql:57]: @DennisGibz did this to handle some data quality on PatientPKHash. worth investigating 
`select * from ODS.dbo.CT_AllergiesChronicIllness
where PatientPKHash is null`

**Review by Marymary-dev — APPROVED** [review]
This is looking good to me, I like the way you have imputed for screened .You will handle the conflicts

**Comment by DennisGibz** [comment]
Checked @nobert-mumo and resolved. I'll also keep on checking to ensure that the update on the PKsHash doesn't fail.
From the merge above I see some conflicts with the merge done. Maybe you can resolve the conflict and I plan to pull the latest changes for analytic refresh

**Comment by nobert-mumo** [comment]
> Checked @nobert-mumo and resolved. I'll also keep on checking to ensure that the update on the PKsHash doesn't fail. From the merge above I see some conflicts with the merge done. Maybe you can resolve the conflict and I plan to pull the latest changes for analytic refresh

Thanks, I have handled the merge conflicts

**Pull request merged**